src/gobstuf/auth/routes.py

```diff
@@ -0,0 +1,127 @@
+import re
+
+from flask import g, request, url_for
+
+from gobcore.secure.config import AUTH_PATTERN, REQUEST_ROLES, REQUEST_USER
+
+from gobstuf.config import INSECURE_PATH
+
+REQUIRED_ROLE_PREFIX = 'fp_'
+MKS_USER_KEY = 'MKS_GEBRUIKER'
+MKS_APPLICATION_KEY = 'MKS_APPLICATIE'
+
+
+def secure_route(rule, func):
+    """
+    Secure routes are protected by gatekeeper
+
+    The headers that are used to identify the user and/or role should be present
+
+    :param func:
+    :return:
+    """
+    def wrapper(*args, **kwargs):
+        # Check that the endpoint is protected by gatekeeper and check access
+        if request.headers.get(REQUEST_USER) and request.headers.get(REQUEST_ROLES) and \
+                _allows_access(rule, *args, **kwargs):
+            return func(*args, **kwargs)
+        else:
+            return "Forbidden", 403
+
+    wrapper.__name__ = f"secure_{func.__name__}"
+    return wrapper
+
+
+def _get_roles():
+    """
+    Gets the user roles from the request headers
+    """
+    try:
+        return [h for h in request.headers.get(REQUEST_ROLES, "").split(",") if h]
+    except AttributeError:
+        return []
+
+
+def _get_role():
+    # Get the first active role which starts with 'fp_'
+    roles = _get_roles()
+    return next((role for role in roles if role.startswith(REQUIRED_ROLE_PREFIX)), None)
+
+
+def _secure_headers_detected(rule, *args, **kwargs):
+    """
+    Check if any secure headers are present in the request
+
+    :param rule:
+    :param args:
+    :param kwargs:
+    :return:
+    """
+    for header, value in request.headers.items():
+        if re.match(AUTH_PATTERN, header):
+            return True
+    return False
+
+
+def _allows_access(rule, *args, **kwargs):
+    """
+    Check access to paths with variable catalog/collection names
+    """
+    role = _get_role()
+    if role:
+        # When a role is found store the MKS USER and APPLICATION in the global object and allow acces
+        setattr(g, MKS_USER_KEY, role)
```

Comment:
Global security variables sound a bit dodgy. How global is the g variable?

Reply:
See https://flask.palletsprojects.com/en/1.1.x/appcontext/#storing-data

It's global within a context and is lost after the context ends